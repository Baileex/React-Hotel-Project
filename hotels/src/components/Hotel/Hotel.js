import React, { Component } from "react";

class Hotel extends Component {
  render() {
    return (
      <article className="hotel">
        <div className="img-container">
          <img
            width="200"
            src="https://images.pexels.com/photos/258154/pexels-photo-258154.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
            alt="hotel"
          />
          <div className="hotel-info">
            <h3>Name :</h3>
            <h4>
              Rating :
              <div class="star-rating" title="50%">
                <div class="back-stars">
                  <i class="fa fa-star" aria-hidden="true"></i>
                  <i class="fa fa-star" aria-hidden="true"></i>
                  <i class="fa fa-star" aria-hidden="true"></i>
                  <i class="fa fa-star" aria-hidden="true"></i>
                  <i class="fa fa-star" aria-hidden="true"></i>

                  <div class="front-stars">
                    <i class="fa fa-star" aria-hidden="true"></i>
                    <i class="fa fa-star" aria-hidden="true"></i>
                    <i class="fa fa-star" aria-hidden="true"></i>
                    <i class="fa fa-star" aria-hidden="true"></i>
                    <i class="fa fa-star" aria-hidden="true"></i>
                  </div>
                </div>
              </div>
            </h4>

            <h5>
              Facilities :
              <span>
                <i className="fas fa-caret-square-down"></i>
              </span>
              <ul>
                <li>1</li>
              </ul>
            </h5>
          </div>
        </div>
      </article>
    );
  }
}

export default Hotel;
