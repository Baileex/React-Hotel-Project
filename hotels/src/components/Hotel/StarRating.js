import React from 'react';
import StarRatingComponent from "react-star-rating-component";


const StarRating = ({rating}) => {
  return (
    <StarRatingComponent
      name="rate1"
      starCount={5}
      value={rating}
      className="rating"
    />   
  );
};

export default StarRating;