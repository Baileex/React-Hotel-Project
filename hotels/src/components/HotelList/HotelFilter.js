import React from "react";
import { Col, Divider, Checkbox, Radio, Row } from "antd";

const HotelFilter = props => {
  const radioStyle = {
    display: "block",
    height: "30px",
    lineHeight: "30px"
  };
  return (
    <div className="filter-bar">
      <Row>
        <h2>Sort by</h2>
      </Row>
      <Divider />
      <h2>Filter by</h2>
      <Row>
        <Divider orientation="left">Amenities</Divider>
        <Checkbox.Group
          options={props.amenitiesOptions}
          value={props.checkedList}
          onChange={props.onGroupChange}
        />
        <Checkbox onChange={props.onCheck} checked={props.checkAll}>
          Check all
        </Checkbox>
      </Row>
    </div>
  );
};

export default HotelFilter;
