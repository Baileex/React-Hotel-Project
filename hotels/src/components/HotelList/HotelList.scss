@import "../../index.scss";

.hotel-list{
  width: 70vw;
  margiN: 0 auto;
  padding: 2rem 0;
}
