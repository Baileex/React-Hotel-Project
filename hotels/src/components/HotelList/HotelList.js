import React, { Component } from 'react';
import Hotel from "../Hotel/Hotel"

class HotelList extends Component {
  render() {
    return (
      <section className="hotel-list">
        <Hotel/>
      </section>
    );
  }
}

export default HotelList;