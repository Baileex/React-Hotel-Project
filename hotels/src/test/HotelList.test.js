import React from "react";
import { mount, shallow } from "enzyme";
import HotelList from "../components/HotelList/HotelList";
import Hotel from "../components/Hotel/Hotel";
import HotelFilter from "../components/HotelList/HotelFilter";
import { wrap } from "module";

describe("HotelList Component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<HotelList />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders six <Hotel/> components", () => {
    const wrapper = shallow(<HotelList />);
    expect(wrapper.find(Hotel)).toHaveLength(6);
  });
  it("renders the <HotelFilter/> component", () => {
    const wrapper = shallow(<HotelList />);
    expect(wrapper.find(HotelFilter)).toHaveLength(1);
  });
});

describe("HotelFilter Component", () => {
  it("sorts hotels by rating ", () => {
    const wrapper = mount(<HotelList />);
    let filter = wrapper.find(HotelFilter);
    let ascbutton = filter.find(".ant-radio-input").at(0);
    ascbutton.simulate("change");
    expect(wrapper.state().sortBy).toEqual("asc");
    expect(
      wrapper
        .find(Hotel)
        .at(0)
        .find("h3")
        .text()
    ).toEqual("Name : hotelfour");

    let descbutton = filter.find(".ant-radio-input").at(1);
    descbutton.simulate("change");
    expect(wrapper.state().sortBy).toEqual("desc");
    expect(
      wrapper
        .find(Hotel)
        .at(0)
        .find("h3")
        .text()
    ).toEqual("Name : hotelone");
  });
});
