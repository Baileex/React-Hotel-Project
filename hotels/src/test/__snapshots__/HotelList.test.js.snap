// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelFilter Component sorts hotels by rating  1`] = `ReactWrapper {}`;

exports[`HotelList Component renders all hotels 1`] = `ShallowWrapper {}`;

exports[`HotelList Component renders correctly 1`] = `ShallowWrapper {}`;
